src/commonTypes.ts

```diff
@@ -119,17 +122,28 @@ export class CustomError extends Error {
   payload?: unknown;
 }
 
-export type MakePageResult = {
-  pageElement: JSX.Element;
-  pageObject: PageObject;
+// Next files: this are the files necessary to render a Next page
+export type NextPageFiles = {
+  documentFile: NextDocumentFile;
+  appFile: NextAppFile;
+  pageFile: NextPageFile;
 };
 
-export type PageInfo = {
-  pageObject: PageObject;
-  pageData: PageData;
+export type NextErrorPageFiles = {
+  documentFile: NextDocumentFile;
+  appFile: NextAppFile;
+  pageFile: NextErrorFile;
+};
+
+export type NextGenericPageFiles = NextPageFiles | NextErrorPageFiles;
```

Comment:
Same, just a suggestion.

```suggestion
export type NextPageFiles = NextExistingPageFiles | NextErrorPageFiles;
```